sensors/wpa_connection_sensor.js

```diff
@@ -18,18 +18,55 @@ const Sensor = require("./sensor.js");
 const r = require('../util/firerouter.js');
 const event = require('../core/event.js');
 const pl = require('../plugins/plugin_loader.js');
+const rclient = require('../util/redis_manager.js').getRedisClient();
 const sclient = require('../util/redis_manager.js').getSubscriptionClient();
 const exec = require('child-process-promise').exec;
 const platformLoader = require('../platform/PlatformLoader.js');
 const platform = platformLoader.getPlatform();
+const PurplePlatform = require('../platform/purple/PurplePlatform')
 const era = require('../event/EventRequestApi.js');
 const EventConstants = require('../event/EventConstants.js');
 const util = require('../util/util.js');
+const LogReader = require('../util/LogReader')
 const _ = require('lodash');
 
 class WPAConnectionSensor extends Sensor {
 
+  async watchLog(line) {
+    // #define WLAN_REASON_UNSPECIFIED 1
+    if (line.includes('CTRL-EVENT-ASSOC-REJECT status_code=1')) {
+      // ignore reject events within a minute as it could coming from multiple SSIDs
+      const now = Date.now() / 1000
+      if (!this.rejects.length || now - 60 > this.rejects[this.rejects.length-1]) {
+        this.rejects.push(now)
+        this.log.debug('added reject event', this.rejects)
+        while (this.rejects[0] < now - this.config.reject_threshold_time_seconds) {
+          const removed = this.rejects.shift()
+          this.log.debug('removed reject', removed)
+        }
+        if (this.rejects.length >= this.config.reject_threshold_count) {
+          this.log.info('Threshold hit, reloading kernal module ...')
+          // sleep to allow IfPresenceSensor to catch the event
+          await exec('sudo rmmod 88x2cs; sleep 3; sudo modprobe 88x2cs')
```

Comment:
only apply when wifi vendor is 88x2cs, check getWlanVendor() from Platform js file.